src/scvmm/azext_scvmm/tests/latest/test_scvmm_scenario.py

```diff
@@ -151,7 +153,22 @@ def test_scvmm(self):
             'az scvmm vm stop -g {resource_group} --name {vm_name} --skip-shutdown'
         )
 
-        self.cmd('az scvmm vm delete -g {resource_group} --name {vm_name} -y')
+        self.cmd(
+            'az scvmm vm create-checkpoint -g {resource_group} --name {vm_name} --checkpoint-name {checkpoint_name} --checkpoint-description {checkpoint_description}',
+        )
+        alias_sub = self.cmd('az scvmm vm show -g {resource_group} --name {vm_name}').get_output_in_json()
+        checkpoint_id = alias_sub['checkpoints'][0]['checkpointId']
+        self.kwargs.update({'checkpoint_id': checkpoint_id})
+
+        self.cmd(
+            'az scvmm vm restore-checkpoint -g {resource_group} --name {vm_name} --checkpoint-id {checkpoint_id}',
+        )
+
+        self.cmd(
+            'az scvmm vm delete-checkpoint -g {resource_group} --name {vm_name} --checkpoint-id {checkpoint_id}',
+        )
+
+        self.cmd('az scvmm vm delete -g {resource_group} --name {vm_name} --deleteFromHost -y')
```

Comment:
It is suggested to add some checks for the commands.

Reply:
Added checks. Pulled vm states and made checks.